concourse/steps/scan_sources.py

```diff
@@ -63,98 +66,156 @@ def scan_component_with_whitesource(
-                ref=commit_hash,
-            )
-
-            # POST component>
-            ci.util.info('POST project')
-            post_project_object.whitesource_client.post_product(
-                product_token=post_project_object.product_token,
-                component_name=component.name,
-                component_version=post_project_object.component_version,
-                requester_email=requester_mail,
-                extra_whitesource_config=extra_whitesource_config,
-                file=tmp_file,
-            )
-
-        # get all projects
-        ci.util.info('retrieving all projects')
-        projects = client.get_all_projects(product_token=product_token)
-
-        # generate reporting table for console
+    whitesource.util.print_cve_tables(tables=tables)
+
+    if len(notification_recipients) > 0:
+        # generate reporting table for notification
         tables = whitesource.util.generate_reporting_tables(
             projects=projects,
             threshold=cve_threshold,
-            tablefmt='simple',
+            tablefmt='html',
+        )
+
+        # get product risk report
+        ci.util.info('retrieving product risk report')
+        prr = ws_client.get_product_risk_report(product_token=product_token)
+
+        # assemble html body
+        body = whitesource.util.assemble_mail_body(
+            tables=tables,
+            threshold=cve_threshold,
         )
 
-        whitesource.util.print_cve_tables(tables=tables)
-
-        if len(notification_recipients) > 0:
-            # generate reporting table for notification
-            tables = whitesource.util.generate_reporting_tables(
-                projects=projects,
-                threshold=cve_threshold,
-                tablefmt='html',
-            )
-
-            # get product risk report
-            ci.util.info('retrieving product risk report')
-            prr = client.get_product_risk_report(product_token=product_token)
-
-            # assemble html body
-            body = whitesource.util.assemble_mail_body(
-                tables=tables,
-                threshold=cve_threshold,
-            )
-
-            # send mail
-            ci.util.info('sending notification')
-            now = datetime.now()
-            fname = f'{component.name}-{now.year}.{now.month}-{now.day}-product-risk-report.pdf'
-
-            attachment = mail.model.Attachment(
-                mimetype_main='application',
-                mimetype_sub='pdf',
-                bytes=prr.content,
-                filename=fname,
-            )
-
-            whitesource.util.send_mail(
-                body=body,
-                recipients=notification_recipients,
-                component_name=component.name,
-                attachments=[attachment],
-            )
+        # send mail
+        ci.util.info('sending notification')
+        now = datetime.now()
+        fname = f'{product_name}-{now.year}.{now.month}-{now.day}-product-risk-report.pdf'
```

Comment:
firstly: consider inlining (only used once)
secondly: how about `datetime.datetime.now().strftime('%Y-%m-%d')`? Admittedly, this is not actually a lot shorter, and not necessarily easier to read.. so the approach chosen here might be the better choice

Reply:
I just realised (by accident..) that this approach does have one disadvantage: it will not include a leading `9` for months and days < 10. -> either ensure a leading zero, or use strftime, please..